desktop/core/src/desktop/auth/api_authentications.py

```diff
@@ -125,6 +127,30 @@ def _handle_public_key(self, access_token):
 
             return public_key_pem
 
+  def _handle_jku_ha(self):
+    res = None
+
+    key_server_urls = AUTH.JWT.KEY_SERVER_URL.get()
+    if not key_server_urls:
+      return None
+
+    if "," in key_server_urls:
+      # Config format is "['url1', 'url2']" for HA, so we need to clean up and split correctly in list.
+      key_server_urls_list = key_server_urls.replace("%20", "").replace("['", "").replace("']", "").replace("'", "").split(',')
+
+      for jku in key_server_urls_list:
+        try:
+          res = requests.get(jku.rstrip('/'), verify=False)
+        except Exception as e:
+          if 'Name or service not known' in str(e):
+            LOG.warning('JKU %s is not available.' % jku)
+
+        # Check response for None and if response code is successful (200) or authentication needed (401), use that host URL.
+        if (res is not None) and (res.status_code in (200, 401)):
```

Comment:
Can it also be a 404?

Reply:
I don't think so, at-least in my testing I've never seen a 404 from the endpoints. 

Plus 404 can be deceiving with some links actually not be there right?